src/main/java/htsjdk/tribble/gff/SequenceRegion.java

```diff
@@ -41,9 +43,28 @@ void setCircular() {
 
     public boolean isCircular(){return  isCircular;}
 
-    public boolean equals(final SequenceRegion other) {
-        return other.start == start && other.end==end && other.contig.equals(contig) && other.isCircular == isCircular;
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof SequenceRegion)) {
+            return false;
+        }
+
+        final SequenceRegion otherSequenceRegion = (SequenceRegion) other;
+        return otherSequenceRegion.start == start && otherSequenceRegion.end==end && otherSequenceRegion.contig.equals(contig) && otherSequenceRegion.isCircular == isCircular;
     }
 
+    private int computeHashCode() {
```

Comment:
This is cached but the value of isCircular can change which invalidates it.  I think you should either force recalculation of the hashcode when setting isCircular or leave it out of the hash code.